app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/TrackerFeedAdapter.kt

```diff
@@ -260,6 +259,41 @@ class TrackerFeedAdapter @Inject constructor(
         }
     }
 
+    private class TrackerAppsProtectionStateViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        companion object {
+            fun create(parent: ViewGroup): TrackerAppsProtectionStateViewHolder {
+                val inflater = LayoutInflater.from(parent.context)
+                val view = inflater.inflate(R.layout.view_device_shield_activity_apps_protection, parent, false)
+                return TrackerAppsProtectionStateViewHolder(view)
+            }
+        }
+
+        var protectedAppsState: AppsProtectionStateView = view.findViewById(R.id.protectedAppsState)
+        var unProtectedAppsState: AppsProtectionStateView = view.findViewById(R.id.unProtectedAppsState)
+
+        fun bind(
```

Comment:
The code here is moved from the `renderAppsData` in `DeviceShieldActivityFeedFragment.kt`.